packages/ui/__stories__/CheckableSelectField.stories.tsx

```diff
@@ -0,0 +1,253 @@
+import React, { ReactNode, useState } from 'react'
+import { logger } from '@hazelcast/services'
+import { Meta, Story } from '@storybook/react'
+import { GroupedOptionsType } from 'react-select'
+
+import { CheckableSelectField, CheckableSelectProps } from '../src/Select/CheckableSelectField'
+import { SelectFieldOption } from '../src/Select/helpers'
+
+import styles from '../src/Select/SelectField.module.scss'
+import { Form, Formik } from 'formik'
+import { CheckableSelectFieldFormik } from '../src/Select/CheckableSelectFieldFormik'
+
+const options: SelectFieldOption<string>[] = [
+  { value: 'darth_vader', label: 'Darth Vader' },
+  { value: 'luke_skywalker', label: 'Luke Skywalker' },
+  { value: 'obi', label: 'Obi-Wan Kenobi' },
+  { value: 'yoda', label: 'Yoda' },
+  { value: 'han_solo', label: 'Han Solo' },
+  { value: 'boba_fett', label: 'Boba Fett' },
+  { value: 'jar_jar_binks', label: 'Jar Jar Binks' },
+]
+
+export default {
+  title: 'Components/CheckableSelectField',
+  component: CheckableSelectField,
+  parameters: {
+    design: {
+      type: 'figma',
+      url: 'https://www.figma.com/file/8mVm6LTbp2Z0RaWWjTZoft/%F0%9F%90%9DHIVE-Hazelcast-Design-System?node-id=23167%3A4292',
+    },
+  },
+  args: {
+    name: 'characters',
+    label: 'Pick Characters',
+    placeholder: 'One or more characters...',
+    noOptionsMessage: () => 'No characters :-(',
+    options,
+    value: [options[1].value],
+    menuPlacement: 'bottom',
+    onBlur: () => logger.log('blur'),
+  },
+} as Meta<CheckableSelectProps<string>>
+
+const Template: Story<CheckableSelectProps<string>> = ({ value: initialValue, ...args }) => {
+  const [value, setValue] = useState<string[]>(initialValue)
+
+  return (
+    <div style={{ height: 350 }}>
+      <CheckableSelectField<string> {...args} value={value} onChange={setValue} />
+    </div>
+  )
+}
+
+export const Default = Template.bind({})
+
+export const Empty = Template.bind({})
+Empty.args = {
+  value: [],
+}
+
+export const WithError = Template.bind({})
+WithError.args = {
+  error: 'Dark side',
+}
+
+export const Hovered = Template.bind({})
+Hovered.args = {
+  className: styles.hover,
+}
+
+export const Focused = Template.bind({})
+Focused.args = {
+  className: styles.focus,
+}
+
+export const FocusedWithError = Template.bind({})
+FocusedWithError.args = {
+  ...Focused.args,
+  ...WithError.args,
+}
+
+export const Disabled = Template.bind({})
+Disabled.args = {
+  disabled: true,
+}
+
+export const WithHelperText = Template.bind({})
+WithHelperText.args = {
+  helperText: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+}
+
+export const Open = Template.bind({})
+Open.args = {
+  menuIsOpen: true,
+}
+
+export const OpenWithInputValue = Template.bind({})
+OpenWithInputValue.args = {
+  ...Empty.args,
+  ...Open.args,
+  inputValue: 'obi',
+}
+
+export const OpenWithInputValueAndNoResults = Template.bind({})
```

Comment:
When there is an input value, some options are filtered out and I click 'Select All' also the hidden options are selected. This doesn't feel right
![image](https://user-images.githubusercontent.com/13898415/127178969-55f35b3c-5575-4bab-ba30-028196294e1e.png)